lib/services/matchers.go

```diff
@@ -140,53 +142,47 @@ type ResourceSeenKey struct{ name, addr string }
 // is not provided but is provided for kind `KubernetesCluster`.
 func MatchResourceByFilters(resource types.ResourceWithLabels, filter MatchResourceFilter, seenMap map[ResourceSeenKey]struct{}) (bool, error) {
 	var specResource types.ResourceWithLabels
+	if resource == nil {
```

Comment:
This check was to satisfy a test, although I don't believe it'll ever actually happen. I didn't want to remove the test either but I couldn't figure out an unsupported type. I will revisit this in the morning and update the test and remove this check

Reply:
`TestMatchResourceByFilters/unsupported_resource_kind`, its send `null` to check. I ended up just added some other random resource like `ConnectionDiagnostic` and i can remove this. Thanks for the reminder